javalib/src/main/scala/java/util/concurrent/ThreadLocalRandom.scala

```diff
@@ -105,8 +105,9 @@ class ThreadLocalRandom extends Random {
   def nextDouble(least: Double, bound: Double): Double = {
     if (least >= bound)
       throw new IllegalArgumentException()
-
-    nextDouble() * (bound - least) + least
+    // Ensure that rangeSize != Double.PositiveInfinity - Issue #2144
+    val rangeSize =  Math.min(bound - least, Double.MaxValue)
+    nextDouble() * rangeSize + least
```

Comment:
The JavaDoc for `Random` in JDK 8 contains an implementation hint for `doubles()`: https://docs.oracle.com/javase/8/docs/api/java/util/Random.html#doubles-double-double-

It does seem to do the stupid `bound - least`. However, it has a magical `nextDown` call in some cases.